### ISSUE TITLE
By use the feature MariaDB get error "Access denied for user 'root'@'localhost'"

### ISSUE THREAD

**Issue by p4rad0xus** [laravel/homestead#1824]
Please note that the Homestead issue tracker is reserved for bug reports and enhancements. We are not always able to debug Vagrant, Provider or Operating System issues, but will do our best to help. Thank you!

## Versions
  - Vagrant: 2.2.19
  - Provider: Virtualbox 6.1
  - Homestead: 13.3.2

## Host operating system
macOS Big Sure 11.7.1

## Homestead.yaml
```yaml
ip: 192.168.56.56
memory: 2048
cpus: 2
provider: virtualbox
authorize: ~/.ssh/id_ed25519.pub
keys:
    - ~/.ssh/id_ed25519
folders:
    -
        map: /Users/p4rad0xus/Developer/github_issue
        to: /home/vagrant/code
sites:
    -
        map: homestead.test
        to: /home/vagrant/code/public
databases:
    - homestead
features:
    -
        mysql: false
    -
        mariadb: true
    -
        postgresql: false
    -
        ohmyzsh: false
    -
        webdriver: false
services:
    -
        enabled: [mysql]
name: github-issue
hostname: github-issue
```

## Vagrant destroy & up output
vagrant up
https://gist.github.com/p4rad0xus/411623f39b83aaf449f7c85e207a282a#file-vagrant_up

vagrant destroy
https://gist.github.com/p4rad0xus/9f0e5bd46f844cd0d9cc8b17553332fb#file-vagrant_destroy

## Expected behavior
The command `vagrant up` should proceed successful and create the named database in the Homestead.yaml.

## Actual behavior
The script stoped at the point when the database should be created with the error "Access denied for user 'root'@'localhost'".

## Steps to reproduce
1. Create a new Homestead project
2. Set the feature `mysql` to `false` and `mariadb` to `true`
3. Run `vagrant up`

## MySQL-User-Table

It looks like a password for root@localhost is never set.

MariaDB [(none)]> Select User, Host, Password from mysql.user;
```
+-------------+-----------+-------------------------------------------+
| User        | Host      | Password                                  |
+-------------+-----------+-------------------------------------------+
| mariadb.sys | localhost |                                           |
| root        | localhost | invalid                                   |
| mysql       | localhost | invalid                                   |
| root        | 0.0.0.0   | *14E65567ABDB5135D0CFD9A70B3032C179A49EE7 |
| homestead   | 0.0.0.0   | *14E65567ABDB5135D0CFD9A70B3032C179A49EE7 |
| homestead   | %         | *14E65567ABDB5135D0CFD9A70B3032C179A49EE7 |
+-------------+-----------+-------------------------------------------+
```

**Comment by fdglefevre** [comment]
It should be fixed by #1817 

# Workaround

```bash
$ vagrant ssh
vagrant@myproject:~$ mysql
MariaDB [(none)]> GRANT ALL ON *.* TO root@'localhost' IDENTIFIED BY 'secret' WITH GRANT OPTION;
MariaDB [(none)]> flush privileges;
MariaDB [(none)]> exit
vagrant@myproject:~$ exit
$ vagrant up --provision
```

**Comment by svpernova09** [comment]
If you're still having issues `mysql_secure_installation` is another workaround:

```
root@homestead:/home/vagrant# mysql_secure_installation

NOTE: RUNNING ALL PARTS OF THIS SCRIPT IS RECOMMENDED FOR ALL MariaDB
      SERVERS IN PRODUCTION USE!  PLEASE READ EACH STEP CAREFULLY!

In order to log into MariaDB to secure it, we'll need the current
password for the root user. If you've just installed MariaDB, and
haven't set the root password yet, you should just press enter here.

Enter current password for root (enter for none):
OK, successfully used password, moving on...

Setting the root password or using the unix_socket ensures that nobody
can log into the MariaDB root user without the proper authorisation.

You already have your root account protected, so you can safely answer 'n'.

Switch to unix_socket authentication [Y/n] n
 ... skipping.

You already have your root account protected, so you can safely answer 'n'.

Change the root password? [Y/n] Y
New password:
Re-enter new password:
Password updated successfully!
Reloading privilege tables..
 ... Success!


By default, a MariaDB installation has an anonymous user, allowing anyone
to log into MariaDB without having to have a user account created for
them.  This is intended only for testing, and to make the installation
go a bit smoother.  You should remove them before moving into a
production environment.

Remove anonymous users? [Y/n] n
 ... skipping.

Normally, root should only be allowed to connect from 'localhost'.  This
ensures that someone cannot guess at the root password from the network.

Disallow root login remotely? [Y/n] n
 ... skipping.

By default, MariaDB comes with a database named 'test' that anyone can
access.  This is also intended only for testing, and should be removed
before moving into a production environment.

Remove test database and access to it? [Y/n] n
 ... skipping.

Reloading the privilege tables will ensure that all changes made so far
will take effect immediately.

Reload privilege tables now? [Y/n] Y
 ... Success!

Cleaning up...

All done!  If you've completed all of the above steps, your MariaDB
installation should now be secure.

Thanks for using MariaDB!
root@homestead:/home/vagrant# mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 42
Server version: 10.10.2-MariaDB-1:10.10.2+maria~ubu2004 mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]>
MariaDB [(none)]>```
